### DISCUSSION TITLE
what is the options of plugins in defineConfig

### DISCUSSION THREAD

**Discussion by ChelesteWang** [egoist/tsup#563] [Q&A]
@egoist I hope you can take the question to answer my question.
I see the type of defineConfig find the option of plugins , I'd like to learn something about how can I set the plugin option ，and what is different points of plugins and esbuildPlugins

```js
/**
     * TSUP plugins
     * @experimental
     * @alpha
     */
    plugins?: Plugin[];

```

**Comment by daolou** [comment]
I tried this config, it seem throw Error:

```
Error: Plugin context is not set
    at PluginContainer.getContext
```

Then I check source code, it real exist a logic error:
1. `buildStarted` will check `context`
![image](https://user-images.githubusercontent.com/22312092/192956818-9c66e41e-8c6b-4214-9491-a4cfee298c56.png)

2. `setContext` method happend in `runEsbuild`:
![image](https://user-images.githubusercontent.com/22312092/192956947-ffa30ced-dae4-4d4a-a5f6-79dd7bf98c46.png)

3. `buildStarted` invoke before `runEsbuild`, while exec buildStarted, it not yet set context, so it should throw error: "Plugin context is not set"
![image](https://user-images.githubusercontent.com/22312092/192957001-5c5140f2-7139-4348-9b81-8a80ca29bc52.png)

**Comment by daolou** [comment]
#735
